### DISCUSSION TITLE
insert statements get id

### DISCUSSION THREAD

**Discussion by woif00** [mkleehammer/pyodbc#1162] [Q&A] [answered]
Hello,

I wanted to ask if it's possible to get the id of a row after inserting without doing a select after?
Like if it's possible to get the id of the inserted row for the cursor?

I couldn't find anything in the docs.
Thanks in advance.

Also, keywords for other users who might search this topic:
"INSERT INTO"; inserted id;

**Comment by gordthompson** [comment, marked as answer]
If your DBMS supports an `OUTPUT` clause (or similar) for DML then you can use that. For example, with MS SQL Server:

```python
import pyodbc

cnxn = pyodbc.connect(
    "DSN=mssql_199;UID=scott;PWD=tiger^5HHH", autocommit=True
)
crsr = cnxn.cursor()

table_name = "discussion_1162"
crsr.execute(f"DROP TABLE IF EXISTS {table_name}")
crsr.execute(
    f"CREATE TABLE {table_name} (id int identity primary key, txt varchar(50))"
)

sql = f"""\
INSERT INTO {table_name} (txt) OUTPUT inserted.id VALUES (?)
"""
new_id = crsr.execute(sql, "foo").fetchval()
print(new_id)  # 1
new_id = crsr.execute(sql, "bar").fetchval()
print(new_id)  # 2
```

**Reply by MulleOne** [reply]
@gordthompson this worked for me, thank you so much.

**Reply by nathant1234567** [reply]
Thanks so much
